language-support/ts/daml-ledger/index.ts

```diff
@@ -517,12 +536,30 @@ class Ledger {
       } else {
         console.error('Ledger.streamQuery unknown message', json);
       }
-    });
+    };
+    const onClose = ({ code, reason }: {
+        wasClean: boolean; code: number;
+        reason: string; target: WebSocket;
+    }): void => {
+      emitter.emit('close', {code, reason});
+      const now = new Date().getTime();
+      // we try to reconnect if we could connect previously and we were live for at least
+      // 'reconnectThreshold'.
```

Comment:
I thought a bit more about this. I don't think we want to have been live for at least X seconds but rather have been connected for X seconds. Imagine your load balancer kills all connections after 1 minute and you set this threshold to 30 seconds. If it takes 40 seconds to go live because there's quite a lot of data coming from the ACS, we would not reconnect. But I would definitely consider a situation worthy of reconnecting.

Since this is a non-breaking improvement, let's do it in a follow-up PR though.